registry.go

```diff
@@ -5,13 +5,33 @@ import (
 	"sync"
 )
 
-var Tasks taskRegistry
+var Tasks TaskMap
 
-type taskRegistry struct {
+type TaskRegistry interface {
```

Comment:
Do you need this interface? I.e. do you create your own task registry implementation? If not lets now over-complicate this...

Reply:
> do you create your own task registry implementation?

It's certainly possible, for example a database-backed task store

 but mostly I prefer structures that ask for what they minimally need from types. It's immediately clear this way that all a `Queue` needs from a task registry is the getter method - not `Register`, `Unregister`, `Range`, or `Reset`. So I'd argue this is, in some sense, less complicated. Personally I'd quite like it to take `func(name String) *Task` rather than an interface at all.

It also mirrors the interfaces for `Storage` and `Redis` you can provide. Seems odd that `Storage` is open to other implementations but this one does not.

Obviously using `TaskRegistry` gets me what I am after in terms of functionality so I defer to your judgement. Let me know and I'll remove.